Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -119,6 +120,15 @@ bool NetworkStorageSession::shouldBlockThirdPartyCookiesButKeepFirstPartyCookies
     return m_registrableDomainsToBlockButKeepCookiesFor.contains(registrableDomain);
 }
 
+void NetworkStorageSession::setCookie(const URL& firstParty, const Cookie& cookie, IsPartitionedCookie isPartitionedCookie)
+{
+    if (!isOptInCookiePartitioningEnabled() || isPartitionedCookie != IsPartitionedCookie::Yes)
+        setCookie(cookie);
```

Comment:
should there be a `return` after this?  or is the idea that if partitioning is disabled or the specific cookie is not partitioned then we'd set it both generally and in the partition?